generate-workflow.py

```diff
@@ -323,92 +324,173 @@ def __str__(self):
         return textwrap.indent(part1 + part2, 6 * ' ')
 
 
+class ActionStep:
```

Comment:
Use `NamedTuple` and/or factor out the duplicated code with `Step` (both if possible; if not, then we'll have to pick one).